### PR TITLE
[EN-3819] New Redux actions

### PR DESCRIPTION
**Pull request by suzubara**
## Description

This PR adds the new Redux actions and corresponding saga for handling form data updates, so that when an input changes, its corresponding subsection _and_ error/valid status can all be updated within one action. It adds a new reducer (`form`) with a slightly modified data structure, where keys correspond to the subsection unique keys (added in https://github.com/18F/e-QIP-prototype/pull/1771), `data` represents the form data for that subsection, errors is an array of error keys (not fully detailed until EN-3835 is completed), and `valid` is a boolean. Example:
```
state.form = {
  IDENTIFICATION_NAME: {
    data: { Name: { ... } },
    errors: [],
    valid: true,
  }
}
```

In this PR, nothing is calling the new `handleSubsectionUpdate` action yet, so even though the new code and reducer have been added, it is not yet in use. The next steps are to complete EN-3835, and to modify each subsection component so that `handleSubsectionUpdate` is called on input change (EN-3850).

## Checklist for Reviewer

- [ ] Review code changes
- [ ] Review changes for Database effects
- [ ] Verify change works in IE browser

More details about this can be found in [docs/review.md](docs/review.md)

### CODE REVIEW

File: src/reducers/form.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+const defaultState = {}
+
+const form = (state = defaultState, action) => {
+  switch (action.type) {
+    case UPDATE_SUBSECTION: {
+      // This rewrites the entire subsection object - use with caution
+      const { key, subsection } = action
+      return {
+        ...state,
+        [`${key}`]: subsection,
```
Comment (christopherhuii) [inline src/reducers/form.js]:
```
[key]: subsection,
```

Does that work?

Comment (suzubara) [reply src/reducers/form.js]: yuuup it does! changing

File: src/sagas/form.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+} from 'redux-saga/effects'
+import { HANDLE_SUBSECTION_UPDATE } from 'constants/actionTypes'
+import { updateSubsection } from 'actions/FormActions'
+import { validateSection } from 'helpers/validation'
+import { selectSubsection, formTypeSelector } from './selectors'
+
+// This fn currently unused. Will be useful if form starts field-level updates.
+export const updateSectionData = (prevData, field, data) => ({
+  ...prevData,
+  [`${field}`]: {
```
Comment (christopherhuii) [inline src/sagas/form.js]: found one more!

**Comment by christopherhuii** [comment]
looks good to me! although the saga stuff is a little bit over my head; haven't worked with sagas much outside of `react-boilerplate`.

**Comment by ryanhofdotgov** [comment]
@suzubara Is this good to merge?

**Pull request merged**